assets/js/googlesitekit/api/middleware/preloading.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Custom preloading middleware.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies.
+ */
+import { getStablePath } from '@wordpress/api-fetch/build/middlewares/preloading';
+
+/**
+ * createPreloadingMiddleware
+ *
+ * Based on preloadMiddleware from from @wordpress/api-fetch, this middle is a single-use per-endpoint and provides cached
+ * data for the first request only and any subsequent requests hit the server.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object} preloadedData Preloaded data paths.
+ * @return {Function} Function.
+ */
+function createPreloadingMiddleware( preloadedData ) {
+	const cache = Object.keys( preloadedData ).reduce( ( result, path ) => {
+		result[ getStablePath( path ) ] = preloadedData[ path ];
+		return result;
+	}, {} );
+
+	let cacheHasExpired = false;
+	return ( options, next ) => {
+		const { parse = true } = options;
+		const uri = options.path;
+		if ( cacheHasExpired ) {
+			return next( options );
+		}
+		setTimeout( () => {
+			cacheHasExpired = true;
+		}, 1000 );
```

Comment:
Let's swap these around as `parse` and `uri` aren't relevant for the expiration logic.
```suggestion
		if ( cacheHasExpired ) {
			return next( options );
		}
		setTimeout( () => {
			cacheHasExpired = true;
		}, 1000 );
		
		const { parse = true } = options;
		const uri = options.path;
		
```